crates/polars-stream/src/physical_plan/lower_group_by.rs

```diff
@@ -520,6 +520,192 @@ fn try_build_streaming_group_by(
     Some(out)
 }
 
+#[expect(clippy::too_many_arguments)]
+pub fn try_build_sorted_group_by(
+    input: PhysStream,
+    keys: &[ExprIR],
+    aggs: &[ExprIR],
+    output_schema: Arc<Schema>,
+    maintain_order: bool,
+    options: Arc<GroupbyOptions>,
+    apply: Option<PlanCallback<DataFrame, DataFrame>>,
+    expr_arena: &mut Arena<AExpr>,
+    phys_sm: &mut SlotMap<PhysNodeKey, PhysNode>,
+    expr_cache: &mut ExprCache,
+    ctx: StreamingLowerIRContext,
+    are_keys_sorted: bool,
+) -> Option<PolarsResult<PhysStream>> {
+    let input_schema = phys_sm[input.node].output_schema.as_ref();
+
+    if keys.is_empty()
+        || apply.is_some()
+        || options.is_rolling()
+        || options.is_dynamic()
+        || (!are_keys_sorted && maintain_order)
+        || keys.iter().any(|k| {
+            k.dtype(input_schema, expr_arena)
+                .is_ok_and(|dtype| dtype.contains_unknown())
+        })
+    {
+        return None;
+    }
+
+    let mut input = input;
+    let mut input_column = unique_column_name();
+    let mut projected = false;
+    let mut row_encoded: Option<Vec<Field>> = None;
+
+    if keys.len() > 1
+        || match keys[0].dtype(input_schema, expr_arena) {
+            Ok(v) => v.is_nested(),
+            Err(err) => return Some(Err(err)),
+        }
+    {
+        let key_fields = match keys
+            .iter()
+            .map(|k| k.field(input_schema, expr_arena))
+            .collect::<PolarsResult<Vec<_>>>()
+        {
+            Ok(v) => v,
+            Err(e) => return Some(Err(e)),
+        };
+        let expr = AExprBuilder::function(
+            keys.to_vec(),
+            IRFunctionExpr::RowEncode(
+                key_fields.iter().map(|k| k.dtype().clone()).collect(),
+                RowEncodingVariant::Unordered,
```

Comment:
Either we must use `Ordered` here I believe, or we must change the `search_sorted` implementation in the node.